pkg/manifestclient/mutation_directory_reader.go

```diff
@@ -0,0 +1,139 @@
+package manifestclient
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/yaml"
+)
+
+func ReadMutationDirectory(mutationDirectory string) (*AllActionsTracker[FileOriginatedSerializedRequest], error) {
+	ret := NewAllActionsTracker[FileOriginatedSerializedRequest]()
+	errs := []error{}
+
+	for _, action := range sets.List(AllActions) {
+		actionDir := filepath.Join(mutationDirectory, string(action))
+
+		currResourceList, err := readSerializedRequestsFromActionDirectory(action, actionDir)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("unable to read %q content in %q: %w", action, actionDir, err))
+		}
+		ret.AddRequests(currResourceList...)
+	}
+
+	return nil, nil
+}
+
+func readSerializedRequestsFromActionDirectory(action Action, mustGatherDir string) ([]FileOriginatedSerializedRequest, error) {
+	currResourceList := []FileOriginatedSerializedRequest{}
+	errs := []error{}
+	err := filepath.WalkDir(mustGatherDir, func(currLocation string, currFile fs.DirEntry, err error) error {
+		if err != nil {
+			errs = append(errs, err)
+		}
```

Comment:
OK, if an error related to this path is reported, we continue walking rather than stopping...